spec/ruby/language/reserved_keywords.rb

```diff
@@ -0,0 +1,113 @@
+require_relative '../spec_helper'
+
+describe "Ruby's reserved keywords" do
+  # Copied from Prism::Translation::Ripper
+  keywords = [
+    "alias",
+    "and",
+    "begin",
+    "BEGIN",
+    "break",
+    "case",
+    "class",
+    "def",
+    "defined?",
+    "do",
+    "else",
+    "elsif",
+    "end",
+    "END",
+    "ensure",
+    "false",
+    "for",
+    "if",
+    "in",
+    "module",
+    "next",
+    "nil",
+    "not",
+    "or",
+    "redo",
+    "rescue",
+    "retry",
+    "return",
+    "self",
+    "super",
+    "then",
+    "true",
+    "undef",
+    "unless",
+    "until",
+    "when",
+    "while",
+    "yield",
+    "__ENCODING__",
+    "__FILE__",
+    "__LINE__"
+  ]
+
+  invalid_kw_param_names = [
+    "BEGIN",
+    "END",
+    "defined?",
+  ]
+
+  invalid_method_names = [
+    "BEGIN",
+    "END",
+    "defined?",
+  ]
+
+  def expect_syntax_error(ruby_src)
```

Comment:
TODO: move to `spec_helper`?

Reply:
I don't think definition of global methods in example files have good effects.